kin-sdk/IKeyStoreProvider.cs

```diff
@@ -0,0 +1,30 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using kin_base;
+
+namespace kin_sdk
+{
+    public interface IKeyStoreProvider
+    {
+        Task AddAccount(KeyPair keypair, Dictionary<string, object> extras = null);
+        // Add account to storage
+
+        Task RemoveAccount(string identifier);
+        // Remove account from storage
+
+        Task<KeyPair> GetAccountByAddress(string PublicAddress);
+        // Get account by public address
+
+        Task<KeyPair> GetAccountAtIndex(int Index);
```

Comment:
This can be defined to be more generic and instead of using keypair I'd create an interface defining what actually needs to be stored

 Task<KeyPair> GetAccountAtIndex<TKey>(TKey key);
 Task RemoveAccount<TKey>(TKey key);